### DISCUSSION TITLE
Can't save payment method for future payments with checkoutComplete mutation when using Stripe

### DISCUSSION THREAD

**Discussion by brunovisnic** [saleor/saleor#10427] [Q&A]
https://docs.saleor.io/docs/3.x/developer/available-plugins/stripe#using-stored-payment-method

When trying to call the _checkoutComplete_ mutation with _paymentData_ parameter set to a JSONString value just like in a example in the link above, I get the following error message.

Any ideas how to fix this?

![image](https://user-images.githubusercontent.com/111509305/185394644-0c14fc84-c0c4-47b7-af61-c64a1351da9b.png)

**Comment by korycins** [comment]
Hey @brunovisnic 
Can you provide more context here? 
- What version of Saleor do you use?
- Can you paste your mutation from the screenshot as text?

From the screenshot, it doesn't look like the issue related to the Stripe plugin, but more like graphql layer. But we need more context here as I just checked this with main branch and I didn't get any issue.

**Reply by brunovisnic** [reply]
Saleor version: 3.5.2

```
mutation {
  checkoutComplete(
    id: "Q2hlY2tvdXQ6MWQxMGRkZjItMWM5MC00YmVmLWI5ZGQtNDM4YzdlOWIyMzhj",
    paymentData: "{\"payment_method_id\": \"pm_1LYQRCHJrmj8rvJTxGsCOt3B\", \"off_session\": false}"
  ) {
    order {
      id
      userEmail
      created
    }
    confirmationNeeded
    confirmationData
    errors {
      field
      message
      code
    }
  }
}
```

**Reply by korycins** [reply]
@brunovisnic Thanks for providing content. It looks like it is a bug in 3.5. Thanks for reporting this. I also tested it with 3.6, the issue doesn't happen there.

**Reply by korycins** [reply]
@brunovisnic 3.5.12 should solve your problem: https://github.com/saleor/saleor/releases/tag/3.5.12
